### DISCUSSION TITLE
Set Expiry date from an ACF Field/ Support for Custom Field

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/publishpress-future#364] [Feature Requests]
We could add support for other sources of expiry date:

> Is it possible to connect post expirator to work with acf date field?
> 
> If you select date from ACF date picker field, it saves it as a post expiration format with default action like moves to another category or to draft or what ever.

**Comment by stevejburge** [comment]
https://wordpress.org/support/topic/expire-by-custom-acf-meta-date-and-time-picker/

> I’m using WordPress to manage a music venue events schedule, and it has two date components. One is the actual post creation date, and the second is the “event date” saved as ACF Custom meta date and time picker, which is in the future. The data is saved in this format, for sorting reasons: 2022-02-24 20:00:00
> 
> This allows me to order events by event date, not post creation date. Well, the site has gotten MASSIVE, and the staff works quickly and would be nice to automate a category change from “upcoming events” to “past events”.
> 
> The plugin allows you to expire based on a Default Date/Time Duration, would it be possible to expire the post based on custom meta like I am using, +1 day? Would it be possible with a custom function? Has anyone done this?

**Comment by stevejburge** [comment]
One more on this: https://wordpress.org/support/topic/set-expire-date-time-using-custom-field/

**Comment by rizaardiyanto1412** [comment]
One more idea 

> Is it possible to have Future look at a custom field and archive based upon that evaluation? For instance, I have a calendar made using ACF. There is an event date. What I want to do is have the cron job run at 1:00am and if today's date is greater than the event date, archive the post.

**Comment by stevejburge** [comment]
> I'm trying to use WP all import to grab an XML file and feed it into Wordpress, Can Future read a custom field on a post to get that data for post expiration?

**Comment by andergmartins** [comment]
This is implemented in the legacy post expiration feature, but we can also add this to the workflow editor. In the workflow editor we need to add support for metadata, or expressions in the Schedule's source date setting.

**Comment by andergmartins** [comment]
Related to #440

**Comment by andergmartins** [comment]
We've implemented this for workflows by using custom metadata and relative date calculation.
